src/sherpa_ai/utils.py

```diff
@@ -322,3 +331,249 @@ def verify_numbers_against_source(text_to_test: str, source_text: str):
         message = f"Don't use the numbers {joined_numbers} to answer the question. Instead, stick to the numbers mentioned in the context."
         return False, message
     return True, None
+
+
```

Comment:
It seems the code related to entity validation is very long. Maybe considering move them into the entity validation so it does not make the utils.py overwhelming.